iroha/src/config.rs

```diff
@@ -127,21 +164,78 @@ impl ConfigurationBuilder {
                 .max_faulty_peers
                 .unwrap_or_else(|| DEFAULT_MAX_FAULTY_PEERS.to_string())
                 .parse()
-                .unwrap_or_else(|_| panic!("Failed to parse {}", MAX_FAULTY_PEERS)),
-        }
+                .map_err(|e| format!("Max faulty peers parse failed: {}", e))?,
+            public_key: self.public_key,
+            private_key: self.private_key,
+        })
     }
 }
 
+/// Parses string formatted as "[ byte1, byte2, ... ]" into `crypto::PublicKey`.
+fn parse_public_key(public_key_string: &str) -> Result<PublicKey, String> {
+    let vector: Vec<u8> = public_key_string
+        .trim_start_matches('[')
+        .trim_end_matches(']')
+        .splitn(32, ',')
```

Comment:
With `splitn` I think we will get `Failed to parse byte` if the key has more than 32 symbols. Instead I think it's better to use simple `split` without number, as the size will be checked when we do `try_into`. This way we will not have hardcoded numbers and the message error will also be correct.